### PR TITLE
python: Added URL class

### PR DESCRIPTION
**Pull request by pyrito**
This PR adds a URL class that maintains `pathlib`-like semantics to manage paths and such.

### CODE REVIEW

File: python/katana/url.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+
+class URL:
+    """
+    URL class that maintains semantics similar to pathlib. This class does not
+    intend to act as a universal class for all path related things! Users can use
+    the class in the following ways:
+
+    - URL("file:///home/") / "a" / "b" => URL("file:///home/a/b")
+    - URL("file:///home/", "a", "b") => URL("file:///home/a/b")
```
Comment (arthurp) [inline python/katana/url.py]: I'm still not a fan of this second constructor. Why do you want to have it?

**Review by arthurp — APPROVED** [review]
Looks good over all. Just make sure you get the tests in here.

File: python/katana/url.py (hunk 1 above)
Comment (pyrito) [reply python/katana/url.py]: I was mimicking `pathlib` since they have something similar with how `Path` is constructed.

Comment (arthurp) [reply python/katana/url.py]: Oh do they? WTF. The python standard library has some really odd stuff in it. My instinct is that good API is more important than exactly paralleling lesser known features of another library. But it's up to you.

Comment (pyrito) [reply python/katana/url.py]: That's a good point. Less is more

**Review by ddn0 — APPROVED** [review]

**Pull request merged**